src/main/java/com/gradle/ccud/adapters/MojoMetadataProviderAdapter.java

```diff
@@ -0,0 +1,124 @@
+package com.gradle.ccud.adapters;
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.MojoExecution;
+import org.apache.maven.project.MavenProject;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * @see com.gradle.develocity.agent.maven.api.cache.MojoMetadataProvider
+ * @see com.gradle.maven.extension.api.cache.MojoMetadataProvider
+ */
+public interface MojoMetadataProviderAdapter {
```

Comment:
@jprinet could you give this PR a review to ensure that the Quarkus Caching Maven extension will be compatible?